packages/web3torrent/src/library/client-seeding.test.ts

```diff
@@ -78,32 +77,24 @@ describe('Seeding and Leeching', () => {
         expect(torrentPeers[`${leecher.pseAccount}`].allowed).toEqual(false);
         done();
       });
-      leecher.add(seededTorrent.magnetURI, {store: MemoryChunkStore});
+      leecher.leech(seededTorrent.magnetURI, {store: MemoryChunkStore});
     });
   }, 10000);
 
-  it.skip('should be able to unchoke and finish a download', done => {
+  it('should unchoke when seeder unblocks leecher', done => {
     seeder.seed(defaultFile as File, defaultSeedingOptions(), seededTorrent => {
-      seeder.once(ClientEvents.PEER_STATUS_CHANGED, ({peerAccount}) => {
-        seeder.togglePeer(seededTorrent.infoHash, peerAccount);
-
+      seeder.once(ClientEvents.PEER_STATUS_CHANGED, async ({peerAccount}) => {
         seeder.once(ClientEvents.PEER_STATUS_CHANGED, ({torrentPeers}) => {
           expect(torrentPeers[`${leecher.pseAccount}`].allowed).toEqual(true);
+          done();
         });
-
-        seeder.on(ClientEvents.TORRENT_NOTICE, ({command}) => {
-          if (command == PaidStreamingExtensionNotices.ACK) {
-            leecher.once(ClientEvents.TORRENT_DONE, ({torrent: leechedTorrent}) => {
-              expect(seededTorrent.files[0].done).toEqual(leechedTorrent.files[0].done);
-              expect(seededTorrent.files[0].length).toEqual(leechedTorrent.files[0].length);
-              expect(seededTorrent.files[0].name).toEqual(leechedTorrent.files[0].name);
-              done();
-            });
-          }
-        });
+        // Add a delay to prevent race condition. Without the delay, leecher may
+        // transfer fund before it joins the channel.
+        await new Promise(r => setTimeout(r, 2000));
```

Comment:
Since we have both `leecher` and `seeder` in context, might there be some promise we could plug into and `await` as opposed to adding a delay?

Reply:
Actually we don't need that. The race condition was caused by the fake channel client sending a wrong turn, somehow causing the leecher to get ahead of itself. Removed.